guti/template/Operation.php

```diff
@@ -0,0 +1,36 @@
+<?php
+abstract class Operation {
+    protected $value1;
+    protected $value2;
+    protected $operation = "";
+    protected $result;
+    protected $round = false;
+
+    final public function execute($valueOne, $valueTwo) {
+        $this->getOperation();
+        $this->setValues($valueOne, $valueTwo);
+        $this->compute();
+        
+        if ($this->round)
+            $this->rounding();
+
+        $this->showResult();
+    }
+
+    public function setValues($val1, $val2) {
+        $this->value1 = $val1;
+        $this->value2 = $val2;
+    }
+
+    public function getOperation() {
+        echo "OPERACIÓN: ".$this->operation."<br>";
+    }
+
+    public function rounding() {
```

Comment:
Lo que hacen tus subclases realmente es sobreescribir la variable $this->result en Operacion, sin embargo, el patron indica que hay que sobreescribir el metodo hook, en este caso rounding()... podrias modificarlo para que las subclases puedan decidir a cuantos decimales redondear? Ejemplo: por default redondear a 2 decimales y a Division darle la habilidad de redondear a 0 decimales